lib/grpc/GrpcService.ts

```diff
@@ -434,6 +455,32 @@ class GrpcService {
     }
   }
 
+  /**
+   * See [[Service.listTrades]]
+   */
+  public listTrades: grpc.handleUnaryCall<xudrpc.ListTradesRequest, xudrpc.ListTradesResponse> = async (call, callback) => {
+    try {
+      const trades = await this.service.listTrades(call.request.toObject());
+      const response = new xudrpc.ListTradesResponse();
+      const tradesList: xudrpc.Trade[] = [];
+      trades.forEach(async (trade: TradeInstance) => {
+        const grpcTrade = new xudrpc.Trade();
+        const makerOrder = await trade.getMakerOrder();
+        const takerOrder = await trade.getTakerOrder();
+        grpcTrade.setQuantity(trade.quantity);
+        grpcTrade.setRHash(trade.rHash ? trade.rHash : '');
+        grpcTrade.setMakerOrder(getGrpcOrderFromOrderInstance(makerOrder!));
+        grpcTrade.setTakerOrder(getGrpcOrderFromOrderInstance(takerOrder!));
```

Comment:
It's possible we won't know of the taker order (if, for example, a peer filled one of our orders via a swap) so instead of using the `!` operator we should put this line within an `if (takerOrder)` statement.